FEAT : ODYA-235 여행일지 등록 api 연결

leeJongMin58 opened the pull request:
### 개요
여행일지 등록 api 연결

### 변경사항

### 관련 지라 및 위키 링크
https://weit.atlassian.net/browse/ODYA-235?atlOrigin=eyJpIjoiNjQ1ZjUyYjMxZjVjNGIyMjhhNTIyNjhjMjg4ZWQ5ZDgiLCJwIjoiaiJ9

### 리뷰어에게 하고 싶은 말
리뷰 작성 api만 연결한거라서 사진이나 자세한 설명은 안붙힐게요

Commit:
FEAT : TravelJounal 생성 api 연결 ver1

특이사항
1. include에 tabID가 없어요
https://lwamuhaji.tistory.com/57

2. 아직 이미지 처리를 못했어요

3. dataStroe도 연결해야 될거 같아요

4. 열심히 하겠습니다.

Commit:
FEAT : 여행일지 등록 성공

Commit:
chore : 종료/등록 모달 및 질문 사항 제외 완성

Commit:
feat : 여행일지 등록하기

- 위도 / 경도 목록은 몰라서 못했습니다..

overthename commented on presentation/src/main/java/com/weit/presentation/ui/post/travellog/PostTravelLogViewModel.kt:
```diff
@@ -160,8 +177,62 @@ class PostTravelLogViewModel @Inject constructor() : ViewModel() {
+                        placeId = log.place?.placeId,
+                        latitudes = emptyList(),
+                        longitudes = emptyList(),
+                        travelDate = listOf(log.date!!.year, log.date.monthValue, log.date.dayOfMonth),
```
!!이거 꼭 써야되나??

minseonglove replied:
log.date가 무조건 nullable임이 보장되고 있나요?

overthename reviewed (approved):
수고하셨습니당 post_visibility 를 include로 한게 다른곳에서도 사용하기 위함인가요?

minseonglove commented on data/src/main/java/com/weit/data/repository/journal/TravelJournalRepositoryImpl.kt:
```diff
@@ -68,6 +69,7 @@ class TravelJournalRepositoryImpl @Inject constructor(
                 } catch (e: Exception) {
                     return Result.failure(e)
                 }
+                Log.d("jomi", "filesName : $fileName")
```
로그 제거

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/post/travellog/PostTravelLogViewModel.kt:
```diff
@@ -56,9 +66,16 @@ class PostTravelLogViewModel @Inject constructor() : ViewModel() {
-            .slice(0 until DEFAULT_FRIENDS_SUMMARY_COUNT)
-            .map { it.profile }
+
+        val friendsSummary = if (travelFriends.size >= DEFAULT_FRIENDS_SUMMARY_COUNT) {
+            travelFriends
+                .slice(0 until DEFAULT_FRIENDS_SUMMARY_COUNT)
+                .map { it.profile }
+        } else {
+            travelFriends
+                .map { it.profile }
+        }
+
```
slice할때 DEFAULT_FRIENDS_SUMMARY_COUNT대신 min(DEFAULT_FRIENDS_SUMMARY_COUNT, travelFriends.size)도 괜찮을거 같음

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/post/travellog/PostTravelLogViewModel.kt:
```diff
@@ -160,8 +177,62 @@ class PostTravelLogViewModel @Inject constructor() : ViewModel() {
+                travelDurationDays = dailyTravelLog.count(),
+                contentImageNameTotalCount = dailyTravelLog.sumOf { it.images.count() }
+            )
+            val travelJournalImages = emptyList<String>().toMutableList()
+
+            for (log in dailyTravelLog){
+                log.images.forEach { image ->
+                    travelJournalImages.add(image)
+                }
+            }
```
travelJournalImages에 log.images를 그대로 할당하는대신 for로 초기화한 이유가 있나요

leeJongMin58 replied:
log.images를 그대로 할당하면 List<List<image>> 꼴이 되기 떄문에 List<image>를 만들기 위해서 for문을 사용했습니다

minseonglove reviewed (commented):
다른건 괜찮은데 !!를 쓴 부분만 확인해주세요

Commit:
feat : 리뷰 수정

1. 주석 및 로그 제거
2. log.date nullable 체크
3. 등록 알럿 설정
4. min(DEFAULT_FRIENDS_SUMMARY_COUNT, travelFriends.size) 추가

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/post/travellog/PostTravelLogViewModel.kt:
```diff
@@ -253,10 +264,17 @@ class PostTravelLogViewModel @Inject constructor(
+        viewModelScope.launch {
+            _event.emit(Event.NoDateInLog(day))
+        }
+    }
```
메소드 이름이 모호하네요 InputDate를 no 한다는건지 InputDate가 no일때 부른단건지 애매함니다

minseonglove commented on presentation/src/main/java/com/weit/presentation/ui/post/travellog/PostTravelLogViewModel.kt:
```diff
@@ -201,33 +204,41 @@ class PostTravelLogViewModel @Inject constructor(
+                            return@launch
+                        } else {
+                            listOf(log.date.year, log.date.monthValue, log.date.dayOfMonth)
+                        },
```
지금도 문제 없지만 객체 초기화 매개변수에 로직을 넣는건 가독성에 좋지 않아요. 밖에서 따로 변수로 초기화 하고 집어넣어주는게 좋아 뵙니다

minseonglove reviewed (approved):
조습니다

Commit:
feat : 수정 2차

1. noInputDate 메서드 이름 변경
2. log.date의 null 로직의 확인 변경

Commit:
Merge branch 'develop' into feature/ODYA-235

# Conflicts:
#	data/src/main/java/com/weit/data/repository/journal/TravelJournalRepositoryImpl.kt
#	presentation/src/main/java/com/weit/presentation/ui/feed/post/FeedPostFragment.kt
#	presentation/src/main/java/com/weit/presentation/ui/memory/MemoryFragment.kt
#	presentation/src/main/res/layout/fragment_memory.xml
#	presentation/src/main/res/navigation/main_graph.xml

Commit:
FEAT :develop브런치 pull

Commit:
Merge branch 'develop' into feature/ODYA-235

# Conflicts:
#	presentation/src/main/java/com/weit/presentation/ui/journal/travel_journal/TravelJournalFragment.kt
#	presentation/src/main/res/values/dimens.xml

Commit:
FEAT : 좌표 입력 기능 추가
